src/libraries/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -657,14 +657,13 @@ void ReadRowOrCol(int startExclusive, int endExclusive, StdInReader reader, Read
                         result = row - 1;
                     }
                 }
+            }
 
-                void TransferBytes(ReadOnlySpan<byte> src, StdInReader dst)
+            static void TransferBytes(ReadOnlySpan<byte> src, StdInReader dst)
+            {
+                for (int i = 0; i < src.Length; i++)
                 {
-                    for (int i = 0; i < src.Length; i++)
-                    {
-                        byte b = src[i];
-                        dst.AppendExtraBuffer(&b, 1);
-                    }
+                    dst.AppendExtraBuffer(src.Slice(i, 1));
```

Comment:
It would be good to follow-up subsequently to understand whether we can just get rid of this whole TransferBytes method, or whether it really is important to add each byte one by one.

Reply:
The motivation seems to be [rolling back an unsuccessful cursor position query](https://github.com/dotnet/corefx/commit/3aac19549465452c36f57ed3749ac8ee73e1956b#diff-5468165269572f0f5bbcc85a2ba0f9c559de23459a54de60bc842904dfd0d4baR393-R397), when the user might have typed their own inputs concurrently. The byte-by-byte copying logic seems deliberate, perhaps to emulate user input? But it feels wrong to me, especially when the console uses non-ascii encodings.